aws/src/main/java/org/apache/iceberg/aws/LakeFormationAwsClientFactory.java

```diff
@@ -0,0 +1,221 @@
+        .name(tableName)
+        .build());
+    boolean isRegisteredWithLakeFormation = response.table().isRegisteredWithLakeFormation() != null ?
+        response.table().isRegisteredWithLakeFormation() : Boolean.FALSE;
+
+    if (isRegisteredWithLakeFormation) {
+      return new LakeFormationCredentialsProvider(properties);
+    } else {
+      return getAwsCredentialsProvider();
+    }
+  }
+
+  private static Region region(Map<String, String> properties) {
+    String region = properties.containsKey(AwsProperties.LAKE_FORMATION_AWS_REGION) ?
+        properties.get(AwsProperties.LAKE_FORMATION_AWS_REGION) : null;
+    if (region == null || region.isEmpty()) {
+      throw new RuntimeException("Region is empty");
+    }
+    return Region.of(region);
+  }
+
+  private AwsCredentialsProvider getAwsCredentialsProvider() {
+    return StaticCredentialsProvider.create(AwsSessionCredentials.create(accessKeyId, secretAccessKey, sessionToken));
+  }
+
+  static class LakeFormationCredentialsProvider implements AwsCredentialsProvider {
+    private LakeFormationClient lakeFormationClient;
+
+    private String tableArn;
+    private static final List<String> privilegeList = Arrays.asList("SELECT", "INSERT", "DELETE");
+
+    LakeFormationCredentialsProvider(Map<String, String> properties) {
+      AwsSessionCredentials awsSessionCredentials = getAssumeRoleCredentials(properties);
+      String region = properties.containsKey(AwsProperties.LAKE_FORMATION_AWS_REGION) ?
+          properties.get(AwsProperties.LAKE_FORMATION_AWS_REGION) : null;
+
+      lakeFormationClient = buildLakeFormationClient(awsSessionCredentials.accessKeyId(),
+          awsSessionCredentials.secretAccessKey(), awsSessionCredentials.sessionToken(), region);
+      tableArn = getTableArn(properties);
+    }
+
+    @Override
+    public AwsCredentials resolveCredentials() {
+      GetTemporaryGlueTableCredentialsRequest getTemporaryGlueTableCredentialsRequest =
+          GetTemporaryGlueTableCredentialsRequest.builder()
+              .tableArn(tableArn)
+              .permissionsWithStrings(privilegeList)
+              .build();
+      GetTemporaryGlueTableCredentialsResponse response =
+          lakeFormationClient.getTemporaryGlueTableCredentials(getTemporaryGlueTableCredentialsRequest);
+      return AwsSessionCredentials.create(response.accessKeyId(), response.secretAccessKey(), response.sessionToken());
+    }
+
+    private AwsSessionCredentials getAssumeRoleCredentials(Map<String, String> properties) {
+      String roleArn = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_ARN);
+      Preconditions.checkNotNull(roleArn, "Cannot initialize LakeFormationAwsClientFactory with empty roleArn");
+      String sessionName = String.format("iceberg-aws-%s", UUID.randomUUID());
+      String externalId = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_EXTERNAL_ID);
+      String tagString = properties.get(AwsProperties.CLIENT_ASSUME_ROLE_TAGS);
```

Comment:
tags property should be named `lakeformation.role.tags.`, and it should be accessed like a prefix, similar to the recent #4259 feature we added. By doing that we can leverage the same util method and no need to parse tag string using `key=xxx,Value=yyy`.

Also, is this a single tag? Based on the API it seems to accept multiple tags, so the prefix way should work better.